core/src/main/java/org/bitcoinj/core/LockTime.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright by the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bitcoinj.core;
+
+import org.bitcoinj.base.internal.TimeUtils;
+
+import java.time.Instant;
+import java.util.Objects;
+
+/**
+ * Wrapper for transaction lock time, specified either as a block height or as a timestamp (in seconds
+ * since epoch). Both are encoded into the same long "raw value", as used in the Bitcoin protocol.
+ * The lock time is said to be "not set" if its raw value is zero.
+ * <p>
+ * Instances of this class are immutable and should be treated as Java
+ * <a href="https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/doc-files/ValueBased.html#Value-basedClasses">value-based</a>.
+ */
+public class LockTime {
+    /**
+     * Wrap a raw value (as used in the Bitcoin protocol) into a lock time.
+     * @param rawValue raw value to be wrapped
+     * @return wrapped value
+     */
+    public static LockTime of(long rawValue) {
+        return new LockTime(rawValue);
+    }
+
+    /**
+     * Wrap a block height into a lock time.
+     * @param blockHeight block height to be wrapped
+     * @return wrapped block height
+     */
+    public static LockTime ofBlockHeight(int blockHeight) {
+        if (blockHeight >= THRESHOLD)
```

Comment:
We should probably prohibit negative values here.  (And probably zero, requiring explicit use of `unset()`)

Reply:
Good idea, done. I excluded the `0` case for now because of backwards compatibility.